main.tf

```diff
@@ -1,16 +1,53 @@
 resource "aws_dx_hosted_private_virtual_interface" "private_vif" {
-  provider         = "aws.source"
+  count = "${var.create_dx_private_hosted_vif ? 1 : 0}"
+
   address_family   = "${var.address_family}"
   bgp_asn          = "${var.bgp_asn}"
   connection_id    = "${var.connection_id}"
   name             = "${var.name}"
   owner_account_id = "${var.owner_account_id}"
   vlan             = "${var.vlan_id}"
-  amazon_address   = "${var.amazonAddress}"
+  amazon_address   = "${var.amazon_address}"
+  mtu              = "${var.mtu}"
 }
 
 resource "aws_dx_hosted_private_virtual_interface_accepter" "private_vif_accepter" {
+  count = "${var.create_dx_private_hosted_vif ? 1 : 0}"
+
   provider             = "aws.accepter"
   virtual_interface_id = "${aws_dx_hosted_private_virtual_interface.private_vif.id}"
-  vpn_gateway_id       = "${var.vgw_id}"
+  vpn_gateway_id       = "${concat(element(aws_vpn_gateway.this.*.id, 0), list(var.vgw_id))}"
```

Comment:
Does this work? vpn_gateway_id only takes a string, not a list. don't you need element(,0) on the outside of it all?

Reply:
Fixed in latest commit.